mimesis/providers/date.py

```diff
@@ -255,3 +254,128 @@ def timestamp(self, posix: bool = True, **kwargs) -> Union[str, int]:
             return timegm(stamp.utctimetuple())
 
         return stamp.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+class FutureDatetime(BaseDataProvider):
+    """Class for generating data related to the date and time in the future."""
+
+    def __init__(self, days: int = 1, weeks: int = 0, hours: int = 0,
+                 minutes: int = 0, seconds: int = 0, *args, **kwargs):
+        """Initialize attributes.
+
+        :param days: Number of days to add to the current date.
+        :param weeks: Number of weeks to add to the current date.
+        :param hours: Number of hours to add to the current date.
+        :param minutes: Number of minutes to add to the current date.
+        :param seconds: Number of seconds to add to the current date.
+        """
+        super().__init__(*args, **kwargs)
+        self.future = datetime.now() + timedelta(
+            days=days, weeks=weeks, hours=hours, minutes=minutes,
+            seconds=seconds)
+        self._dt = Datetime(*args, **kwargs)
+        self.day_of_week = self._dt.day_of_week
+        self.month = self._dt.month
+        self.century = self._dt.century
+        self.periodicity = self._dt.periodicity
+        self.time = self._dt.time
+        self.formatted_time = self._dt.formatted_time
+        self.day_of_month = self._dt.day_of_month
+        self.timezone = self._dt.timezone
+        self.gmt_offset = self._dt.gmt_offset
+        self.timestamp = self._dt.timestamp
```

Comment:
@lk-geimfari @sobolevn I had to switch to composition instead of inheritance because of the incompatibility with the overloaded methods.

I know this section is a bit tricky. It can be replaced with:

```
        self._dt = Datetime(*args, **kwargs)
        for method in (
            m for m in dir(Datetime)
            if callable(getattr(Datetime, m))
            and not m.startswith('_')
            and m not in dir(FutureDatetime)
        ):
            setattr(self, method, getattr(self._dt, method))
```
which it automatically adds the methods not overridden, but it is more black magic and less explicit. What do you prefer?

Reply:
@ceccoemi I think that we need to implement only some methods in this provider, not all of them. We must decide which methods can be really useful in `FutureDatetime`.

Reply:
@lk-geimfari ok, if you don't want all the methods then it's really easy.

I propose to keep `date`, `formatted_date`, `datetime`, `formatted_datetime` and `timestamp` only.

Reply:
@lk-geimfari see my last commit!

Reply:
@ceccoemi I agree with your propose.